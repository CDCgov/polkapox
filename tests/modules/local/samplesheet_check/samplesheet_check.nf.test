nextflow_process {

    name "Test Process SAMPLESHEET_CHECK"
    script "modules/local/samplesheet_check/samplesheet_check.nf"
    process "SAMPLESHEET_CHECK"

    test("Validate samplesheet") {

        when {
            params {

            }
            process {
                """
                 input[0] = file("${projectDir}/assets/samplesheet.test.csv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
