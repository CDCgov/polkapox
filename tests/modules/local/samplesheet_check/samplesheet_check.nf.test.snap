{
    "Validate samplesheet": {
        "content": [
            {
                "0": [
                    "samplesheet.valid.csv:md5,736ba5ae4db834d3aa0bcead2c7a3ea1"
                ],
                "1": [
                    "versions.yml:md5,18be61d02e0cddb04ad768625a029906"
                ],
                "csv": [
                    "samplesheet.valid.csv:md5,736ba5ae4db834d3aa0bcead2c7a3ea1"
                ],
                "versions": [
                    "versions.yml:md5,18be61d02e0cddb04ad768625a029906"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:10:00.755954244"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "samplesheet.valid.csv:md5,a2dd7c0c201ac9c59ab433e83cbb8173"
                ],
                "1": [
                    "versions.yml:md5,18be61d02e0cddb04ad768625a029906"
                ],
                "csv": [
                    "samplesheet.valid.csv:md5,a2dd7c0c201ac9c59ab433e83cbb8173"
                ],
                "versions": [
                    "versions.yml:md5,18be61d02e0cddb04ad768625a029906"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T12:24:17.155878561"
    }
}